### PR TITLE
LSAS controls

### PR DESCRIPTION
**Pull request by ritika-msft**
## Description
</br>
 Brief description of the requested change.

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1 [hunk 1]
```diff
@@ -2156,5 +2156,351 @@ class SubscriptionCore: AzSVTBase
...
+	}
+
+	hidden [ControlResult] CheckApplicationsWithoutOwners([ControlResult] $controlResult)
+	{
+		$tenantId = ([ContextHelper]::GetCurrentRMContext()).Tenant.Id
+		$allValidRAs = $null
+		$allPrivRAs = $null
+		$allPrivSPNRAs = $null
+		
+        $allRAsSubScope = Get-AzRoleAssignment -Scope "/subscriptions/$($this.SubscriptionContext.SubscriptionId)"
```
Comment (TarunKrShukla) [inline src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1:2194]: We should not be calling Get-AzRoleAssignment for same scope in each control, better we cache it in class variable.

Comment (ritika-msft) [reply src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1]: Added class variable and made the required changes. Now Get-AzRoleAssignment will not be called if cached data is found.

File: src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1 [hunk 2]
```diff
@@ -31,11 +31,13 @@ class SubscriptionCore: AzSVTBase
 	hidden $SubscriptionExtId;
 	hidden [boolean] $IsSubAssignmentFetched = $false;
 	hidden [boolean] $IsRGAssignmentFetched = $false;
+	hidden [PSObject] $allRAsSubScope;
```
Comment (Aboli-msft) [inline src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1:34]: initialize to null

File: src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1 [hunk 3]
```diff
@@ -2156,5 +2158,367 @@ class SubscriptionCore: AzSVTBase
 		}
 		return $controlResult
 	}
+
+	hidden [ControlResult] CheckAccessOfDeletedObjects([ControlResult] $controlResult)
+	{
+		# Get all role assignments
+        if ($this.allRAsSubScope -eq $null)
+		{
+			$this.allRAsSubScope = Get-AzRoleAssignment -Scope "/subscriptions/$($this.SubscriptionContext.SubscriptionId)"
```
Comment (Aboli-msft) [inline src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1:2167]: Check if classic roles need to be fetched

Comment (Aboli-msft) [reply src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1:2167]: Also check if CA is able to fetch details for roles

File: src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1 [hunk 4]
```diff
@@ -2156,5 +2158,367 @@ class SubscriptionCore: AzSVTBase
...
+		{
+			$this.allRAsSubScope = Get-AzRoleAssignment -Scope "/subscriptions/$($this.SubscriptionContext.SubscriptionId)"	
+		}
+
+        $allValidRAs = $this.allRAsSubScope | Where-Object {$_.ObjectType -ne 'Unknown'}
+		$this.GetPIMRoles();
+		$pimroles = $this.PIMAssignments #list of Eligible Pim assignments
+		$allValidRAs += $pimroles
+
+        $privilegedRoles = @('Owner','Contributor','User Access Administrator')
```
Comment (Aboli-msft) [inline src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1:2318]: Move to controlsettings

Comment (Aboli-msft) [reply src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1:2318]: check indentation

File: src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1 [hunk 5]
```diff
@@ -2156,5 +2158,367 @@ class SubscriptionCore: AzSVTBase
...
+				}
+			}
+			else
+			{
+				$controlResult.AddMessage([VerificationResult]::Passed, [MessageData]::new("No application(s) found with privileged roles on subscription"));
+			}
+	}
+	catch 
+	{
+		throw [System.ArgumentException] ("Unable to fetch details of owners of privileged application.");
```
Comment (Aboli-msft) [inline src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1:2406]: change the message

File: src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1 (hunk 2 above)
Comment (ritika-msft) [reply src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1]: Intialized variable with null

File: src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1 (hunk 4 above)
Comment (ritika-msft) [reply src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1]: Moved to control Settings, Now we are fetching these details from control settings

File: src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1 (hunk 3 above)
Comment (ritika-msft) [reply src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1]: Used the method defined in Helpers class to fetch role assignments

File: src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1 (hunk 5 above)
Comment (ritika-msft) [reply src/AzSK/Framework/Core/SVT/SubscriptionCore/SubscriptionCore.ps1]: changed

**Review by Aboli-msft — APPROVED** [review]

**Review by TarunKrShukla — DISMISSED** [review]

**Review by TarunKrShukla — APPROVED** [review]

**Review by Aboli-msft — APPROVED** [review]

**Pull request merged**